todo_app/data/trello_items.py

```diff
@@ -0,0 +1,247 @@
+import requests
+import os
+from dotenv import load_dotenv
+from dateutil import parser
+
+class Item:
+
+    def __init__(self, id, name, list_id, description, due_date):
+        self.id = id
+        self.name = name
+        self.idList = list_id
+        self.description = description
+        self.due_date = due_date
+
+    @classmethod
+    def from_trello_cards(cls, card):
+        return cls(card["id"],
+                   card["name"],
+                   card["idList"],
+                   card["desc"],
+                   card["due"])
+
+    def clean_date(self):
+        if self.due_date:
+            trello_date = parser.parse(self.due_date)
+            self.due_date = trello_date.date()
+
+
+def call_api(url, method, query, headers=None):
+    """call the specified url with the specified call and paramaters
+
+    Inputs
+        url: url endpoint to be called
+        params: parameters for the api call
+        method: type of call to make to the url
+
+    Raises
+        Exception in the case of connection error
+
+    Outputs
+        response: json of the response
+    """
+
+    try:
+        response = requests.request(
+            method,
+            url,
+            headers=headers,
+            params=query
+        )
+
+        response.raise_for_status()
+
+        return response.json()
+
+    except requests.exceptions.HTTPError as http_err:
+
+        print(f"HTTP error occurred: {http_err}")
+
+    except Exception as err:
+
+        print(f"Error occurred: {err}")
+
+
+def get_items(board_id, apikey, token):
+    """
+    Get all action items on the trello board
+
+    Inputs
+        board_id: id of the board we want the items for
+        apikey: apikey for accessing trello
+        token: api token for accessing trello
+
+    Returns
+        response_json: a json of the response from the request
+    """
+
```

Comment:
Please can you implement the following helper function and refactor all the methods in this file to use it:

```suggestion
def build_url(endpoint):
    return 'https://api.trello.com/1' + endpoint
```

This means that the base URL is only set once, reducing duplication.